src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1024,68 +1097,74 @@ public async Task<string> GetDebuggerDisplayAttribute(SessionId sessionId, int o
-                        var objectValues = await GetObjectValues(sessionId, objectId, true, false, false, false, token);
 
-                        var thisObj = CreateJObject<string>("", "object", "", false, objectId:$"dotnet:object:{objectId}");
-                        thisObj["name"] = "this";
-                        objectValues.Add(thisObj);
-
-                        var resolver = new MemberReferenceResolver(proxy, context, sessionId, objectValues, logger);
-                        var dispAttrStr = displayAttrValue["value"]?["value"]?.Value<string>();
-                        //bool noQuotes = false;
-                        if (dispAttrStr.Contains(", nq"))
-                        {
-                            //noQuotes = true;
-                            dispAttrStr = dispAttrStr.Replace(", nq", "");
-                        }
-                        expr = "$\"" + dispAttrStr + "\"";
-                        JObject retValue = await resolver.Resolve(expr, token);
-                        if (retValue == null)
-                            retValue = await EvaluateExpression.CompileAndRunTheExpression(expr, resolver, token);
-                        return retValue?["value"]?.Value<string>();
-                    }
-                    catch (Exception)
+                    parmCount = retDebuggerCmdReader.ReadInt32();
+                    for (int j = 0; j < parmCount; j++)
                     {
-                        logger.LogDebug($"Could not evaluate DebuggerDisplayAttribute - {expr}");
-                        return null;
+                        //to named_args
+                        await CreateJObjectForVariableValue(sessionId, retDebuggerCmdReader, "varName", false, -1, token);
                     }
                 }
-                else
+            }
+            return null;
+        }
+        public async Task<string> GetDebuggerDisplayAttribute(SessionId sessionId, int objectId, int typeId, CancellationToken token)
+        {
+            string expr = "";
+            try {
+                var invokeParams = new MemoryStream();
+                var invokeParamsWriter = new MonoBinaryWriter(invokeParams);
+                var retDebuggerCmdReader = await GetCAttrsFromType(sessionId, objectId, typeId, "System.Diagnostics.DebuggerDisplayAttribute", token);
+                if (retDebuggerCmdReader != null)
                 {
+                    invokeParamsWriter.Write((byte)ValueTypeId.Null);
+                    invokeParamsWriter.Write((byte)0); //not used
+                    invokeParamsWriter.Write(0); //not used
                     var parmCount = retDebuggerCmdReader.ReadInt32();
-                    for (int j = 0; j < parmCount; j++)
+                    retDebuggerCmdReader.ReadByte();
+                    var stringId = retDebuggerCmdReader.ReadInt32();
+                    var dispAttrStr = await GetStringValue(sessionId, stringId, token);
```

Comment:
`dispAttrStr` could be null, or empty. The documentation suggests that `an empty string ("") causes the value column to be hidden.`. the hidden stuff doesn't need to be done in this PR though, but handle null/empty.

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggerdisplayattribute.-ctor?view=net-5.0